Upload Code Dailog with a code editor

abhishekchohan opened the pull request:
**What it does ?**
- Implements the Dialog box reusable component.
- User can select the language from dropdown.
- Dialog box contains 1 code editor to edit code (That highlights the selected language syntax)
- Comment box that take user input for comments if any for the added code.
- this PR implements new custom scrollbars throughout the app.
- Some code refactoring..etc..

**What to avoid?**
- Nothing except you will see 2 text editor but one is just a code snippet displayed with syntax highlighting.
- This is done to have a boilerplate code ready for future implementations in other pages.

**Lastly: Navbar can use this reusable component instead of the hard coded button now.**

**Some screenshots**

![component_button](https://user-images.githubusercontent.com/22992889/96257830-1314d280-0f89-11eb-9fdb-c1fb43ecf851.png)

![select langauge](https://user-images.githubusercontent.com/22992889/96257837-160fc300-0f89-11eb-9957-5418a23ff3df.png)

![code editor with highlighter](https://user-images.githubusercontent.com/22992889/96257851-1a3be080-0f89-11eb-952c-c9e97ed10a43.png)

![no-editable-codeSnippet](https://user-images.githubusercontent.com/22992889/96257863-1dcf6780-0f89-11eb-8d1e-d69765e89d41.png)

skassam21 commented on client/src/components/UploadCodeDailog.js:
```diff
@@ -0,0 +1,252 @@
+//modes ==> tells the codemirror library about the kind of
+//file we are opening so that respective syntax highlighting can be achieved.
+
+const modes = [
```
can we extract this into a constants file - as we will probably be using this in multiple places

abhishekchohan replied:
Done as suggested

skassam21 commented on client/src/components/UploadCodeDailog.js:
```diff
@@ -0,0 +1,252 @@
+const modes = [
+  { label: "JavaScript", value: "jsx" },
+  { label: "CSS", value: "css" },
+  { label: "JAVA", value: "java" },
```
Should be "Java"

abhishekchohan replied:
Fixed.

skassam21 reviewed (commented):
Can you make it look closer to the spec? Here are a couple of comments:

1. No need to have the comment div under the Submit button
2. You will need a Title TextField object 
3. Can you add some margin between the comments and the codeMirror?

Did you want to try to add RichText to the textField?

abhishekchohan commented:
> Can you make it look closer to the spec? Here are a couple of comments:
> 
> 1. No need to have the comment div under the Submit button
> 2. You will need a Title TextField object
> 3. Can you add some margin between the comments and the codeMirror, and the selection dropdown?
> 
> Did you want to try to add RichText to the textField?


1. commentDiv was just a placeholder but I removed it.
2. ok sure. Working on it.
3. I added the margins as suggested.

yes I tried that yesterday but found this module little easy so implemented that but no worries, now I will start working on the rich text instead.

Thanks for the feedback
